### DISCUSSION TITLE
Hit error still rounded to 300Hz?

### DISCUSSION THREAD

**Discussion by arvindf232** [ppy/osu#28140] [Q&A] [answered]
I found out that my keyboard seems to be performing at 300Hz: The hit error seems to be grouping up into blocks of 3ms (about 7 groups to both sides at OD10 = +-20ms). This grouping happens when I play maps of 240 or 225 bpm, but I cannot tell for sure if it is happening for other bpm or not. 

I saw this discussion https://github.com/ppy/osu/discussions/19918 from 2 years ago. In that discussion, the keyboard is to blamed. I am using a Wooting HE60 keyboard, which given its reputation, should be reliable and the 1000Hz polling should be true. Of course, there can still be other hardware or OS issues. 

In the discussions, it was said that "It is based on the game's update frame rate for now." I see that the fps counter shows "update frame 300fps" and "input 

1000fps". So, having the hit errors still based on update frame rate could explain the issue. Yet, two years have passed, so I am asking to see this had not been changed and the update frame could still be the reason to blame. 

Attached: Hit error bar. The map is 225bpm and 9.78OD (150bpm and 8OD + DT)

![Screenshot 2024-05-09 at 10 10 37 PM](https://github.com/ppy/osu/assets/34409283/3ecba121-adf1-43d0-a48b-94348f58f00a)

**Comment by peppy** [comment, marked as answer]
Yes, game processing is still coupled to update frame rate currently. This may change in the future but requires timestamping of inputs.

https://github.com/ppy/osu-framework/issues/4808
